lib/main.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:async';
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:pocketape/arkitTracer.dart';
+// ignore: depend_on_referenced_packages
+import 'package:vector_math/vector_math_64.dart';
+
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: MeasurementScreen(),
+    );
+  }
+}
+
+class MeasurementScreen extends StatefulWidget {
+  @override
+  _MeasurementScreenState createState() => _MeasurementScreenState();
+}
+
+class _MeasurementScreenState extends State<MeasurementScreen> {
+  late Stream<Vector3> _stream;
+  Vector3? _firstValue;
+  Vector3? _lastValue;
+  double distance = 0.00;
+  late StreamSubscription<Vector3> _subscription;
+  bool _isMeasuring = false;
+
+  @override
+  void initState() {
+    super.initState();
+    _stream = ArkitTracer.trace();
+     _subscription = _stream.listen(
+        (position) {
+          setState(() {
+            if (_firstValue == null) {
+              _firstValue = position;
+            }
+            else {
+              _lastValue = position;
+            }
+          });
+        },
+        onError: (error) {},
+        onDone: () {
+        },
+      );
+      _subscription.pause();
+  }
+
+  void _toggleMeasurement() {
+    if (_isMeasuring) {
+      _stopMeasurement();
+    } else {
+      setState(() {
+        _firstValue = null;
+        _lastValue = null;
+      });
+      _startMeasurement();
+    }
+  }
+
+  void _startMeasurement() {
+    if (!_isMeasuring) {
+      setState(() {
+        _isMeasuring = true;
+      });
+      _subscription.resume();
+    }
+  }
+
+  void _stopMeasurement() {
+    if (_isMeasuring) {
+      _subscription.pause();
+      setState(() {
+        _isMeasuring = false;
+        if (_firstValue != null && _lastValue != null) {
+          distance = calculateDistance(_firstValue!, _lastValue!);
+        }
+      });
+    }
+  }
```

Comment:
Use `_subscription.cancel()` and set it to `null`. (this way you can also replace `_isMeasuring` with `bool get isMeasuring => _subscription != null`). When you start a new measuring start a new subscription (`_stream.listen`)

Reply:
I used the .pause() and resume() because the stream can only have one listener. If i use cancel() it does not complete since somehow waits for the stream to be clean which never happen.  I request your clarification because I applied your suggestions and it does not work, and still face the problem I tried to avoid with the pause() and resume()